src/pages/ServicePage.js

```diff
@@ -26,28 +26,84 @@ class CategoryPage extends React.Component {
     componentDidMount(): void {
         iss(`service/${this.props.params.id}/`)
             .then(data => {
-                console.log(data);
                 this.setState({
                     object: data,
                 });
             });
 
-        // FIXME: display error
+        // FIXME: display error on failure to connect
+    }
+
+    renderPhones(phones: Array<Object>): React.Element {
+        if (phones && phones.length) {
+            var first = phones.shift();
+            var extras = null;
+            if (phones.length) {
+                extras = (
+                    <Collapser message="Other contact options">
+                        {
+                            phones.map((phone, idx) =>
+                                <Phone {...phone} key={idx} />
+                            )
+                        }
+                    </Collapser>
+                );
+            }
+
+            return (
+                <mui.Paper>
+                    <Phone {...first} />
+                    {extras}
+                </mui.Paper>
+            );
+
+        } else {
+            // No phones
+            return null;
+        }
+
+    }
+
+    eligibility(eligibility:string): Array<React.Element> {
+        eligibility.split("\n").map((line) => <li>{line}</li>)
     }
 
     render(): React.Element {
         var {
             object,
         } = this.state;
-
-        return (
+        if (!object) {
+            return <div/>;
+        } else return (
             <div>
                 <Router.RouteHandler />
-                <mui.AppBar title={object.name} />
+                <mui.AppBar title={object.site.name} />
                 <mui.Paper>
                     <p>
-                        {object.site.name}
+                        {object.name}
                     </p>
+                    <p>
+                        {object.description}
+                    </p>
+                </mui.Paper>
+                <mui.Paper>
+                    <OpeningTimes object={object} />
+                </mui.Paper>
+                <mui.Paper>
+                    <Address {...object.location} />
+                </mui.Paper>
+                {this.renderPhones(object.phones)}
+                <mui.Paper>
+                    <h3>To receive this service you must be</h3>
```

Comment:
Be aware these may not be set. In which case I don't think we should show that particular heading.